### PR TITLE
start locking down no-tenant handling

### PR DESCRIPTION
**Pull request by malaffoon**
Goal is to remove any special consideration of no-tenant, and clean up any non-tenant-aware stuff. I also moved a couple things into common from elsewhere since they are used by both RDW_Ingest and RDW_Reporting.

There are changes in both RDW_Ingest and RDW_Reporting (https://github.com/SmarterApp/RDW_Reporting/pull/1959) that have to be coordinated with this.

### CODE REVIEW

File: archive/src/main/java/org/opentestsystem/rdw/archive/ArchiveServiceTenantDynamicRouter.java [hunk 1]
```diff
@@ -35,8 +36,11 @@ protected ArchiveService archiveServiceFactory(String key) {
     }
 
     private ArchiveService getArchiveService() {
-        final String tenantKey = this.tenantKeyResolver.getTenantKey().orElse(TenantKeyResolver.NO_TENANT_KEY);
-        return tenantArchiveServices.computeIfAbsent(tenantKey, this::archiveServiceFactory);
+        final Optional<String> tenantKey = this.tenantKeyResolver.getTenantKey();
+        if (!tenantKey.isPresent()) {
+            throw new IllegalStateException("No tenant key set");
+        }
```
Comment (malaffoon) [inline archive/src/main/java/org/opentestsystem/rdw/archive/ArchiveServiceTenantDynamicRouter.java]: The general philosophy i've adopted is: Resolvers return Optionals, and the thing that uses the Resolver (router, interceptor, etc.) is what throws the exception.

File: multi-tenant/src/main/java/org/opentestsystem/rdw/multitenant/messaging/TenantMessageRequestResolver.java [hunk 2]
```diff
@@ -2,17 +2,16 @@
...
     /**
-     * Resolves the tenant id
-     *
-     * @param message Message
-     * @return tenantId
+     * @param message message from which to extract the tenant id
+     * @return the tenant id
      */
-    String resolveTenantId(Message<?> message);
+    Optional<String> getTenantId(Message<?> message);
```
Comment (malaffoon) [inline multi-tenant/src/main/java/org/opentestsystem/rdw/multitenant/messaging/TenantMessageRequestResolver.java]: changed to match overall philosophy

File: script/src/main/java/org/opentestsystem/rdw/script/publishing/ArchiveBackedPublishedPipelineRepository.java [hunk 3]
```diff
@@ -63,7 +63,7 @@ public PipelineDefinition findByCodeAndVersion(final String code, final String v
                                     .build()
                     ))
                     .build();
-        } catch (final Exception exception) {
+        } catch (final IllegalArgumentException ignored) {
```
Comment (malaffoon) [inline script/src/main/java/org/opentestsystem/rdw/script/publishing/ArchiveBackedPublishedPipelineRepository.java]: @esotrope, i mentioned this in one of your reviews; i'm going on the assumption that it is a good change.

Comment (esotrope) [reply script/src/main/java/org/opentestsystem/rdw/script/publishing/ArchiveBackedPublishedPipelineRepository.java]: i changed something to illegal argument exception - cant remember what at this point though x l

**Pull request merged**